x-pack/plugins/alerts/server/lib/alert_instance_summary_from_event_log.test.ts

```diff
@@ -271,6 +275,79 @@ describe('alertInstanceSummaryFromEventLog', () => {
       Object {
         "instances": Object {
           "instance-1": Object {
+            "actionGroupId": "action group A",
+            "activeStartDate": "2020-06-18T00:00:00.000Z",
+            "muted": false,
+            "status": "Active",
+          },
+        },
+        "lastRun": "2020-06-18T00:00:10.000Z",
+        "status": "Active",
+      }
+    `);
+  });
+
+  test('alert with currently active instance with no action group in event log', async () => {
+    const alert = createAlert({});
+    const eventsFactory = new EventsFactory();
+    const events = eventsFactory
+      .addExecute()
+      .addNewInstance('instance-1')
+      .addActiveInstance('instance-1', undefined)
+      .advanceTime(10000)
+      .addExecute()
+      .addActiveInstance('instance-1', undefined)
+      .getEvents();
+
+    const summary: AlertInstanceSummary = alertInstanceSummaryFromEventLog({
+      alert,
+      events,
+      dateStart,
+      dateEnd,
+    });
+
+    const { lastRun, status, instances } = summary;
+    expect({ lastRun, status, instances }).toMatchInlineSnapshot(`
+      Object {
+        "instances": Object {
+          "instance-1": Object {
+            "actionGroupId": undefined,
+            "activeStartDate": "2020-06-18T00:00:00.000Z",
+            "muted": false,
+            "status": "Active",
+          },
+        },
+        "lastRun": "2020-06-18T00:00:10.000Z",
+        "status": "Active",
+      }
+    `);
+  });
+
+  test('alert with currently active instance that switched action groups', async () => {
+    const alert = createAlert({});
+    const eventsFactory = new EventsFactory();
+    const events = eventsFactory
+      .addExecute()
+      .addNewInstance('instance-1')
+      .addActiveInstance('instance-1', 'action group A')
+      .advanceTime(10000)
+      .addExecute()
+      .addActiveInstance('instance-1', 'action group B')
```

Comment:
This makes me wonder what kind of events we should be generating when an instance "switches" action groups.  My first thought was it should probably send a `resolved-instance` with the old action group, and a `new-instance` with the new instance group, in between these `active-instance` events.  Which implies sending the action groups on `new-instance` and `resolved-instance` as well.  

But not sure.  Perhaps it would be better to leave `new-instance` and `resolved-instance` as is, and maybe have a new event `active-action-group-changed` or such.

I think we'll need to think about this a little bit - create a new issue?  ie "what events should be logged when an alert switches action groups".

Reply:
Created this issue: https://github.com/elastic/kibana/issues/82792